.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+repos:
+
+# Check if the commit message has a valid sign-off.
+- repo: https://github.com/gklein/check_signoff
+  rev: v1.0.5
+  hooks:
+    - id: check-signoff
+
+# Catch gofmt issues, if any.
+- repo: git://github.com/dnephin/pre-commit-golang
+  rev: master
+  hooks:
+    - id: go-fmt
```

Comment:
this is having may hooks why not add all :D
```
- id: go-fmt
    - id: go-vet
    - id: go-lint
    - id: go-imports
    - id: go-cyclo
      args: [-over=15]
    - id: validate-toml
    - id: no-go-testing
    - id: gometalinter
    - id: golangci-lint
    - id: go-critic
    - id: go-unit-tests
    - id: go-build
    - id: go-mod-tidy
```

Reply:
We might want to keep it minimal as It will run before each and every commit we make. It might become annoying if it takes more than a couple of seconds. Any thoughts?

Reply:
at least we can run basic `go-vet go-lint and go-vet`  will it take more time?

Reply:
Okay, I will test it out locally and verify the delay. If the delay is not drastic, we can definitely add them as well.